wifiphisher/common/interfaces.py

```diff
@@ -217,180 +272,220 @@ def __init__(self):
-        ap_available = list()
-        monitor_available = list()
-
-        # Populate ap_available and monitor_available lists
-        for k, interface in self._interfaces.iteritems():
-            # Add all the interfaces with monitor mode
-            if interface.has_monitor_mode():
-                monitor_available.append(interface)
-            # Add all the interfaces with AP mode
-            if interface.has_ap_mode():
-                ap_available.append(interface)
-
-        # Raise error if no interface with AP mode is found
-        if len(ap_available) == 0:
-            raise NoApInterfaceFoundError()
-        # Raise error if no interface with monitor mode is found
-        if len(monitor_available) == 0:
-            raise NoMonitorInterfaceFoundError()
-        # Raise error if one card is supposed to do both
-        if len(monitor_available) == 1 and len(ap_available) == 1:
-            if monitor_available[0] == ap_available[0]:
-                raise NotEnoughInterfacesFoundError()
-
-        # We only have one AP mode interface. We don't want to use it for
-        # jamming.
-        if len(monitor_available) > 1 and \
-                len(ap_available) == 1 and \
-                ap_available[0] in monitor_available:
-            # Select an AP interface and remove it from available interfaces
-            ap_interface = ap_available[0]
-            ap_available.remove(ap_interface)
-            # Select the first available interface with monitor mode
-            for m in monitor_available:
-                if m != ap_interface:
-                    monitor_interface = m
-            return monitor_interface, ap_interface
-
-        # We only have one Monitor mode interface. We don't want to use it for AP.
-        # Covers all other cases too
-        monitor_interface = monitor_available[0]
-        # Select the first available interface with monitor mode
-        for a in ap_available:
-            if a != monitor_interface:
-                ap_interface = a
-
-        return monitor_interface, ap_interface
-
-    def get_jam_iface(self, interface_name):
-        for k, interface in self._interfaces.iteritems():
-            if k == interface_name and not interface.being_used:
-                if interface.has_monitor_mode():
-                    return interface
-                else:
-                    raise JammingInterfaceInvalidError
-        raise JammingInterfaceInvalidError
-
-    def get_ap_iface(self, interface_name=None):
-        for k, interface in self._interfaces.iteritems():
-            if interface_name == None:
-                if interface.has_ap_mode():
-                    return interface
-            if k == interface_name and not interface.being_used:
-                if interface.has_ap_mode():
-                    return interface
+        """
+
+        monitor_interface = self.get_interface(has_monitor_mode=True)
+        ap_interface = self.get_interface(has_ap_mode=True)
```

Comment:
I thought about it myself but it seems that I have messed up the order. Using your example swapping them should work since if interface has AP it is guaranteed to have monitor mode as well.